configure.ac

```diff
@@ -200,7 +200,7 @@ AC_PATH_PROG(PERL, perl)
 #   -Wall                        CC, CXX
 #
 
-PROSPECTIVE_CFLAGS="-Wall"
+PROSPECTIVE_CFLAGS="-Werror"
```

Comment:
Noticing we have excluded "third_party" code from the check, I think it may be worth to try stricter check on our own code. Any concerns on the potential issues? @jwhui